### PR TITLE
The legend of xDays is back!

### PR DESCRIPTION
**Pull request by FrilledShark**
I have created a new version of my feature xDays. I have used this configuration for about a week. I copied the changes made to Github.

I'm not quite sure what I did. The script has evolved a bit since last time I used it. I just tried and tried until it was stable. I havn't observed any issues with it yet.
Feel free to look at it and see if it fits.

Old pull request: #8

### CODE REVIEW

**Comment by Evanito** [comment]
LGTM! 👍

**Comment by rnevet** [comment]
The change in logic is ok, however, documentation and backward compatibility are not taken care off.

**Comment by FrilledShark** [comment]
I will try to fix backward compatibility. Is it ok if I just check if sixtydaythreshold is still present and if it is, use that instead of xdaythreshold?

**Comment by rnevet** [comment]
Yeah that would be perfect, just don't want to break existing configs. If sixtydaythreshold  is present use it's value for the xdaythreshold with xdays=60?
Otherwise README also needs to be updated with the new config options.

**Comment by FrilledShark** [comment]
I have tried to fix Backwards compatibility on my own branch. I will not put it on this pull request before I have tested it for bugs.
View my changes here: https://github.com/FrilledShark/poloniexlendingbot/commit/4847467f3fb3bcd483830fbdac19a92c7a3ef925

**Comment by FrilledShark** [comment]
I have tried to make it backwards compatible. It works on my local copy, but try to see if I actually make it work.

**Comment by FrilledShark** [comment]
Done.

File: lendingbot.py [hunk 1]
```diff
@@ -132,7 +139,7 @@
 #End handling args.
 
 #Check if we need a config file at all (If all settings are passed by args, we won't)
-if args.apikey and args.apisecret and args.sleeptimeactive and args.sleeptimeinactive and args.mindailyrate and args.maxdailyrate and args.spreadlend and args.gapbottom and args.gaptop and args.sixtydaythreshold:
+if args.apikey and args.apisecret and args.sleeptimeactive and args.sleeptimeinactive and args.mindailyrate and args.maxdailyrate and args.spreadlend and args.gapbottom and args.gaptop and args.xdaythreshold and args.xdays:
```
Comment (rnevet) [inline lendingbot.py]: This means that sixtydaythreshold will not work as command line?

Comment (FrilledShark) [inline lendingbot.py]: I didn't know what to put there, so I just changed it from sixtydaythreshold to xdays...

**Comment by rnevet** [comment]
I would say something like:
'''
((args.xdaythreshold and args.xdays) or args.sixtydaythreshold)
'''

File: lendingbot.py (hunk 1 above)
Comment (rnevet) [inline lendingbot.py]: I would say something like:
```
((args.xdaythreshold and args.xdays) or args.sixtydaythreshold)
```

Comment (FrilledShark) [inline lendingbot.py]: Like this right? https://github.com/Mikadily/poloniexlendingbot/pull/108/commits/de313ae6541bf4187e3be312038ad5462f76031b

Comment (rnevet) [inline lendingbot.py]: I would say yes.

**Comment by rnevet** [comment]
@Evanito, can you test this a bit before we merge it? I won't have access to my dev environment in the upcoming weeks.

**Comment by FrilledShark** [comment]
It was running with my first pull request when I started it. I have updated changes now.

**Comment by rnevet** [comment]
That's great, I would like to have 1 more person running it before merging. I would myself but can't at the moment.

**Pull request merged**